src/model/utils.js

```diff
@@ -0,0 +1,196 @@
+                            // TODO: enable this in the future
+                            // throw new Error(`Casting error, "${item.type}" type not known.`);
+                            console.log(`Casting error, "${item.type}" type not known.`);
+                    }
+                }
+            });
+            newNode = new CompositionNode(item.id, '', false, ports, extra, children);
+            break;
+        }
+        case PNLTypes.MECHANISM: {
+            let ports = [];
+            let extra = {};
+            item.attr_list.forEach(singleAttr => {
+                if (singleAttr.id === 'label') {
+                    // TODO: implement the parsing of the json structure generated below
+                    // in order to detect ports and other elements of the node.
+                    //
+                    // parsedHtml = html2json(singleAttr.eq);
+                    return;
+                }
+                if (singleAttr.type === 'attr') {
+                    extra[singleAttr?.id] = singleAttr?.eq;
+                }
+            });
+            newNode = new MechanismNode(item?.node_id?.id, '', false, ports, extra);
+            break;
+        }
+        case PNLTypes.PROJECTION: {
+            let name = '';
+            let extra = {};
+            let sender, senderPort, receiver, receiverPort;
+            item.attr_list.forEach(singleAttr => {
+                if (singleAttr.id === 'label') {
+                    name = singleAttr.eq;
+                    return;
+                }
+                if (singleAttr.type === 'attr') {
+                    extra[singleAttr?.id] = singleAttr?.eq;
+                }
+            });
+            if (item.edge_list.length === 2) {
+                sender = item.edge_list[0].id;
+                senderPort = item.edge_list[0]['port']['id'];
+                receiver = item.edge_list[1].id;
+                receiverPort = item.edge_list[1]['port']['id'];
+            }
+            newNode = new ProjectionLink(name, sender, senderPort, receiver, receiverPort, false, extra);
+            break;
+        }
+        default:
+            // TODO: enable this in the future
+            // throw new Error(`Casting error, "${item.type}" type not known.`);
+            console.log(`Casting error, "${item.type}" type not known.`);
+    }
+    return newNode;
+}
+
+
+export function buildModel(frontendModel, coord, prevModel) {
+    let finalModel = {
+        mechanisms: [],
+        projections: [],
+        compositions: [],
+    };
+
+    if (prevModel) {
+        finalModel= prevModel;
+    }
+
+    let coordinates = {
+        x: 200,
+        y: 150,
+    };
+
+    let linkCounter = 1;
+
+    if (coord) {
+        coordinates.x = coord.x;
+        coordinates.y = coord.y;
+    }
+
+    frontendModel?.mechanisms?.forEach( node => {
+        if (Array.isArray(node)) {
+            node.forEach( mech => {
+                finalModel.mechanisms.push(
```

Comment:
I was envisioning this to call node.getMetaNode() and let the logic of conversion be implemented by the specialized classes, what do you think?

Reply:
done as well